configapi/api_inventory.go

```diff
@@ -72,52 +107,47 @@ func DeleteGnb(c *gin.Context) {
 }
 
 func handlePostGnb(c *gin.Context) error {
-	var gnbName string
-	var exists bool
-	if gnbName, exists = c.Params.Get("gnb-name"); !exists {
+	gnbName, exists := c.Params.Get("gnb-name")
+	if !exists {
 		errorMessage := "post gNB request is missing gnb-name"
 		logger.ConfigLog.Errorln(errorMessage)
-		return errors.New(errorMessage)
+		return fmt.Errorf("%s", errorMessage)
 	}
 	logger.ConfigLog.Infof("received gNB %v", gnbName)
-	var err error
-	var newGnb configmodels.Gnb
-
-	allowHeader := strings.Split(c.GetHeader("Content-Type"), ";")
-	switch allowHeader[0] {
-	case "application/json":
-		err = c.ShouldBindJSON(&newGnb)
+	if !strings.HasPrefix(c.GetHeader("Content-Type"), "application/json") {
+		return fmt.Errorf("invalid header")
 	}
+	var postGnbRequest configmodels.PostGnbRequest
+	err := c.ShouldBindJSON(&postGnbRequest)
 	if err != nil {
 		logger.ConfigLog.Errorf("err %v", err)
-		return fmt.Errorf("failed to create gNB %v: %w", gnbName, err)
+		return fmt.Errorf("invalid JSON format")
 	}
-	if newGnb.Tac == "" {
+	if postGnbRequest.Tac == "" {
 		errorMessage := "post gNB request body is missing tac"
 		logger.ConfigLog.Errorln(errorMessage)
-		return errors.New(errorMessage)
+		return fmt.Errorf("%s", errorMessage)
+	}
+	postGnb := configmodels.Gnb{
+		Name: gnbName,
+		Tac:  postGnbRequest.Tac,
 	}
-	req := httpwrapper.NewRequest(c.Request, newGnb)
```

Comment:
I see you are removing the `httpwrapper` (github.com/omec-project/util/httpwrapper) from this file. Does it mean that it is "obsolete" and the way you are doing it now (`postGnb`) is better? If so, is this something that should be done across the whole 5GC codebase? For example, the `httpswrapper` is also used in `configapi/api_slice_mgmt.go`, should we follow the same approach there as is done here (configapi/api_inventory.go)?

Reply:
That depends on what you need to do with the new request created by `httpwrapper.NewRequest`. On this case, getting the Gnb using `c.ShouldBindJSON`, then creating a new request using the `httpwrapper` just to extract the Gnb again, seemed unnecessary. Not using the `httpwrapper` is more straightforward. 
If you use the `httpwrapper` to somehow filter the content on the body and use that request for something else again, sure, let's keep it. But if we just need to extract the important content from the body, I think we can remove it.